source/Server.ts

```diff
@@ -0,0 +1,69 @@
+/// <reference path="../typings/node/node.d.ts" />
+var fs = require("fs")
+var http = require("http")
+var url = require("url")
+
+module Exclusive {
+	export class Server {
+		private server: any;
+		constructor(private port: number) {
+			this.server = http.createServer(this.requestCallback);
+		}
+		start() {
+			this.server.listen(this.port, () => {
+				console.log("listening on port " + this.port)
+			});
+		}
+		stop() {
+			this.server.close(() => {
+				console.log("Exclusive server closed")
+			})
+		}
+		createRequest(url: string, request: string) {
+			// TODO: create POST
+		}
+		private requestCallback(request: any, response: any) {
```

Comment:
In the request callback, you need to buffer your data, because there is not guarantee that all of the data is delivered in the same chunk.